ls1intum/Artemis
# Bugfix/exam mode programming exercise status sync

### Checklist

### Motivation and Context


The exam navigation bar and the exam exercise overview page are not in sync, which may confuse students when working on an exam.

### Description

It can happen that the navigation bar shows a different exercise status than the exam exercise overview.

### Steps for Testing


1. Work on an exam programming exercise with the online code editor 
2. Submit the exercise
3. Change something but don't submit it
4. Go to the overview and see that it is not updated properly

The described process that visualizes the bug
![OverviewNotUpdated](https://user-images.githubusercontent.com/63976129/125210414-fac7af00-e29f-11eb-94e5-9b1adf619203.gif)

### Screenshots


Fixed version:
![FixedOverviewNotUpdated](https://user-images.githubusercontent.com/63976129/125215390-aa5e4a80-e2bb-11eb-8700-14f79dd1c02a.gif)
